pkg/resources/secondary_database.go

```diff
@@ -0,0 +1,433 @@
+	},
+	"as_replica_of": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "A fully qualified path to a database to create a replica from. A fully qualified path follows the format of `\"<organization_name>\".\"<account_name>\".\"<database_name>\"`.",
+	},
+	"is_transient": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		ForceNew:    true,
+		Description: "Specifies the database as transient. Transient databases do not have a Fail-safe period so they do not incur additional storage costs once they leave Time Travel; however, this means they are also not protected by Fail-safe in the event of a data loss.",
+	},
+	"data_retention_time_in_days": {
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"value": {
+					Type:     schema.TypeInt,
+					Required: true,
+				},
+			},
+		},
+		Computed:    true,
+		Optional:    true,
+		Description: "Specifies the number of days for which Time Travel actions (CLONE and UNDROP) can be performed on the database, as well as specifying the default Time Travel retention time for all schemas created in the database. For more details, see [Understanding & Using Time Travel](https://docs.snowflake.com/en/user-guide/data-time-travel).",
+	},
+	"max_data_extension_time_in_days": {
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"value": {
+					Type:     schema.TypeInt,
+					Required: true,
+				},
+			},
+		},
+		Computed:    true,
+		Optional:    true,
+		Description: "Object parameter that specifies the maximum number of days for which Snowflake can extend the data retention period for tables in the database to prevent streams on the tables from becoming stale. For a detailed description of this parameter, see [MAX_DATA_EXTENSION_TIME_IN_DAYS](https://docs.snowflake.com/en/sql-reference/parameters.html#label-max-data-extension-time-in-days).",
+	},
+	"external_volume": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		Description:      "The database parameter that specifies the default external volume to use for Iceberg tables.",
+	},
+	"catalog": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		Description:      "The database parameter that specifies the default catalog to use for Iceberg tables.",
+	},
+	"default_ddl_collation": {
+		Type:        schema.TypeString,
```

Comment:
do we want to validate it anyhow or is it the one of the fields that we leave for the SF validation? (I am leaning towards the second option, just confirming)

Reply:
Yeah, I wouldn't validate it.